src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts

```diff
@@ -60,6 +64,22 @@ export class ExerciseDetailsStudentActionsComponent implements OnInit {
                 this.getRepositoryPassword();
             }
         });
+        // In order to check if the application is currently running on production
+        this.profileService.getProfileInfo().subscribe(
+            (profileInfo) => {
+                if (profileInfo) {
+                    this.profileInfo = profileInfo;
+                }
+            },
+            (reason) => {},
+        );
+    }
+
+    /**
+     * The button should be only visible in the dev environment
+     */
+    visibilityCheck(): boolean {
+        return !this.profileInfo.inProduction;
```

Comment:
This is duplicated with the `ProgrammingExerciseUpdateComponent`, maybe you can refactor it to only use it once (maybe create a method `isProdProfile` or something like that in `ProfileService`?

Reply:
I created a method in the profile service and this one will be then called in the particular class